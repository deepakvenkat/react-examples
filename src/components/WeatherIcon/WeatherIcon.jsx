import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

const styles = {
  first: {
    position: 'absolute',
    // clip: rect(top, right, bottom, left)
    backgroundImage: (props) => (`url(\'/public/icons/${props.weather}.png\')`),
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'contain',
    backgroundColor: '#fff4f4',
    height: '100px',
    width: '100px',
  },
  avatar: {
    height: '100px',
    width: '100px',
  },
};

const WeatherIcon = ({
  classes,
}) => (
  <Avatar
    variant="square"
    className={classes.avatar}
  >
    <div className={classes.first}></div>
  </Avatar>
);

WeatherIcon.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  weather: PropTypes.string.isRequired,
};

export default withStyles(styles)(WeatherIcon);
