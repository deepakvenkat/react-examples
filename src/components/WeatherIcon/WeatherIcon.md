## WeatherIcon

```jsx
<WeatherIcon weather="sunny" />
<br />
<WeatherIcon weather="snowfall" />
```
