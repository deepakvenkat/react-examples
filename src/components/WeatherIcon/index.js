import WeatherIcon from './WeatherIcon.jsx';

export default WeatherIcon;
