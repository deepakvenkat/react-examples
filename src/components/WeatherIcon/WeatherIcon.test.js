import React from 'react';
import { render } from '@testing-library/react';
import WeatherIcon from './WeatherIcon';

test('renders the WeatherIcon', () => {
  const { asFragment } = render(
    <WeatherIcon  weather="sunny"/>
  );
  expect(asFragment()).toMatchSnapshot();
});
