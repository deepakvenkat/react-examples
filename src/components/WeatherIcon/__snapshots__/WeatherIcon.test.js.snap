// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the WeatherIcon 1`] = `
<DocumentFragment>
  <div
    class="MuiAvatar-root MuiAvatar-square WeatherIcon-avatar-2 MuiAvatar-colorDefault"
  >
    <div
      class="WeatherIcon-first-1 WeatherIcon-first-3"
    />
  </div>
</DocumentFragment>
`;
